File: magma-api/src/main/java/de/jalumu/magma/platform/MagmaPlatform.java
package de.jalumu.magma.platform;

import de.jalumu.magma.module.ModuleLoader;
import revxrsal.commands.CommandHandler;

import java.util.logging.Logger;

public interface MagmaPlatform {

    String getName();

    String getVersion();

    MagmaPlatformType getPlatformType();

    ServerImplementation getServerImplementation();

    String getPlatformName();

    String getPlatformVersion();

    String getServerID();

    void setServerID(String serverID);

    Logger getLogger();

    CommandHandler getCommandHandler();

    Object getMagmaPluginInstance();

    static MagmaPlatform getPlatform() {
        return MagmaPlatformProvider.getPlatform();
    }


}

File: magma-api/src/main/java/de/jalumu/magma/platform/MagmaPlatform.kt
package de.jalumu.magma.platform

import revxrsal.commands.CommandHandler
import java.util.logging.Logger

/**
 * MagmaPlatform is an Entrypoint for the MagmaKT platform plugins
 */
interface MagmaPlatform {
    val magmaImplementationName: String
    val magmaImplementationVersion: String
    val platformType: MagmaPlatformType
    val serverImplementation: ServerImplementation
    val platformName: String
    val platformVersion: String
    var serverID: String?
    val magmaLogger: Logger
    val commandHandler: CommandHandler
    val magmaPluginInstance: Any

    companion object {
        val platform: MagmaPlatform?
            get() = MagmaPlatformProvider.getPlatform()
    }
}
